nrf_security/Kconfig

```diff
@@ -162,117 +161,189 @@ config VANILLA_MBEDTLS_AES_C
+	select VANILLA_MBEDTLS_CIPHER_MODE_CFB if GLUE_MBEDTLS_AES_C
 	help
 	  Enable the AES Cipher Feedback mode (CFB) mode,
 	  MBEDTLS_CIPHER_MODE_CFB setting in mbed TLS config file.
 
+config GLUE_MBEDTLS_CIPHER_MODE_OFB
+	bool
+
+config CC310_MBEDTLS_CIPHER_MODE_OFB
+	bool
+
+config VANILLA_MBEDTLS_CIPHER_MODE_OFB
+	bool
+
 config MBEDTLS_CIPHER_MODE_OFB
 	bool "AES-OFB - AES Output Feedback mode"
+	depends on MBEDTLS_AES_C || VANILLA_MBEDTLS_AES_C || CC310_MBEDTLS_AES_C
 	default y
-	depends on VANILLA_MBEDTLS_AES_C || (MBEDTLS_VANILLA_BACKEND && MBEDTLS_AES_C)
+	select MBEDTLS_CIPHER_AES_256_OFB_C if GLUE_MBEDTLS_AES_C
+	select GLUE_MBEDTLS_CIPHER_MODE_OFB if GLUE_MBEDTLS_AES_C
+	select CC310_MBEDTLS_CIPHER_MODE_OFB if GLUE_MBEDTLS_AES_C
+	select VANILLA_MBEDTLS_CIPHER_MODE_OFB if GLUE_MBEDTLS_AES_C
 	help
 	  Enable the AES Output Feedback mode (OFB) mode,
 	  MBEDTLS_CIPHER_MODE_OFB setting in mbed TLS config file.
 
 config MBEDTLS_CIPHER_MODE_XTS
 	bool "AES-XTS - AES Xor-encrypt-xor with ciphertext stealing mode"
 	default y
-	depends on VANILLA_MBEDTLS_AES_C || (MBEDTLS_VANILLA_BACKEND && MBEDTLS_AES_C)
+	depends on MBEDTLS_VANILLA_BACKEND && (MBEDTLS_AES_C || VANILLA_MBEDTLS_AES_C)
+	depends on !CC310_BACKEND
 	help
 	  Enable the AES Xor-encrypt-xor with ciphertext stealing mode (XTS) mode,
 	  MBEDTLS_CIPHER_MODE_XTS setting in mbed TLS config file.
 
+config MBEDTLS_AES_256_CMAC_C
+	bool
+
 config MBEDTLS_CMAC_C
 	bool
 	prompt "AES-CMAC - AES Cipher-based Message Authentication Code mode for block ciphers"
-	depends on (MBEDTLS_AES_C || CC310_MBEDTLS_AES_C || VANILLA_MBEDTLS_AES_C )
 	default y
+	depends on (CC310_BACKEND || MBEDTLS_VANILLA_BACKEND)
+	select MBEDTLS_AES_256_CMAC_C if MBEDTLS_VANILLA_BACKEND && !NRF_CRYPTO_BACKEND_COMBINATION_0
 	help
-	  Enable the Cipher-based Message Authentication Code (CMAC) mode for
-	  block ciphers.
+	  Enable the CMAC module.
 	  MBEDTLS_CMAC_C setting in mbed TLS config file.
 
 choice
 	bool
 	prompt "Backend selection"
-	depends on MBEDTLS_CMAC_C
-	depends on NRF_CRYPTO_BACKEND_COMBINATION_0
+	depends on MBEDTLS_CMAC_C && NRF_CRYPTO_BACKEND_COMBINATION_0
 	default CC310_MBEDTLS_CMAC_C
 
-	config CC310_MBEDTLS_CMAC_C
-	    bool "cc310 (CMAC-AES-128)"
-	    depends on CC310_MBEDTLS_AES_C
+config CC310_MBEDTLS_CMAC_C
+	bool "cc310" if CC310_BACKEND
+	help
+	  HW accelerated CMAC support for AES-128
+
+config VANILLA_MBEDTLS_CMAC_C
+	bool "mbed TLS vanilla" if MBEDTLS_VANILLA_BACKEND
+	select MBEDTLS_AES_256_CMAC_C
+	help
+	  CMAC support for AES-128, AES-192, AES-256
 
-	config VANILLA_MBEDTLS_CMAC_C
-	    bool "mbed TLS (CMAC-AES-128, CMAC-AES-192, CMAC-AES-256)"
-	    depends on VANILLA_MBEDTLS_AES_C
 endchoice
 
-endif # AES_CONFIGURATION 
+endmenu # AES   - Advanced Encryption Standard
 
-menuconfig AEAD_CONFIGURATION
-	bool
-	prompt "AEAD  - Authenticated Encryption with Associated Data"
+menu "AEAD  - Authenticated Encryption with Associated Data"
```

Comment:
I think changing this from menuconfig to menu is a step back in usability, as users must now enter the submenu to see if the group is configured or not.
The menuconfig allows user to completely disable AEAD if it is not needed.

Reply:
Fixed after some discussions